### ISSUE TITLE
missing_panics_doc: FP with <div>

### ISSUE THREAD

**Issue by DaniPopes** [rust-lang/rust-clippy#15353]
### Summary

False positive with `<div>` HTML tag. Encountered when adding rustdoc warning to a function (`<div class="warning">`).

### Lint Name

missing_panics_doc

### Reproducer

I tried this code:

```rust
#![warn(clippy::missing_panics_doc)]

/// Summary.
/// <div></div>
/// # Panics
pub fn panics() {
    panic!();
}
```

I saw this happen:

```
warning: docs for function which may panic missing `# Panics` section
 --> src/lib.rs:6:1
  |
6 | pub fn panics() {
  | ^^^^^^^^^^^^^^^
  |
note: first possible panic found here
 --> src/lib.rs:7:5
  |
7 |     panic!();
  |     ^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_panics_doc
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(clippy::missing_panics_doc)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I expected to see this happen: nothing

### Version

```text
rustc 1.90.0-nightly (b56aaec52 2025-07-24)
binary: rustc
commit-hash: b56aaec52bc0fa35591a872fb4aac81f606e265c
commit-date: 2025-07-24
host: x86_64-unknown-linux-gnu
release: 1.90.0-nightly
LLVM version: 20.1.8
```

**Comment by samueltardieu** [comment]
The parsing done by `pulldown-cmark` starts a `HtmlBlock` when it encounters the `<div>` and ends it only at the end of the comment block:

```example
Start(Paragraph)
Text(Borrowed("Summary."))
End(Paragraph)
Start(HtmlBlock)
Html(Borrowed("<div></div>\n"))
Html(Borrowed("# Panics"))
End(HtmlBlock)
```

@notriddle Is that expected?

**Comment by notriddle** [comment]
If you're not sure whether rustdoc (which uses pulldown-cmark) is parsing something correctly, try it in the [commonmark dingus](https://spec.commonmark.org/dingus/?text=Summary.%0A%3Cdiv%3E%3C%2Fdiv%3E%0A%23%20Panics%0A) (which uses commonmark.js).

It does the same thing, so it's not a bug. It might be an opportunity for a diagnostic improvement.
